### PR TITLE
BLD: use directories to reproduce

### PR DESCRIPTION
**Pull request by tylerjereddy**
* `Unix Makefiles` will now once again no longer
be able to avoid rebuilds of consuming modules
with the minor changes in this PR/branch when using
Intel compiler

* this will happen over and over again:

```
cmake -P rebuild.cmake
[ 28%] Built target mod
[ 57%] Built target smod2
[100%] Built target main
```

### CODE REVIEW

**Comment by tylerjereddy** [comment]
this is just for demonstration purposes for the CMake team--so they can easily pull in a branch

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,12 @@ cmake_minimum_required(VERSION 3.14...3.22)
 
 project(rebuild LANGUAGES Fortran)
 
+
+add_subdirectory(mod)
+add_subdirectory(smod2)
```
Comment (bradking) [inline CMakeLists.txt:7]: Please add the subdir `CMakeLists.txt` files to this PR.

**Comment by bradking** [comment]
> [ 28%] Built target mod
> [ 57%] Built target smod2
> [100%] Built target main

With no other output, that is normal output from CMake-generated makefiles even when nothing is rebuilt.

File: CMakeLists.txt (hunk 1 above)
Comment (tylerjereddy) [reply CMakeLists.txt:7]: woops, I think I pushed them in now

**Comment by tylerjereddy** [comment]
> With no other output, that is normal output from CMake-generated makefiles even when nothing is rebuilt.

Argh, so I still don't have a reproducer?

**Comment by bradking** [comment]
With the example in this PR, `touch mod/mod.f90` only causes the one source to recompile.

**Comment by tylerjereddy** [comment]
Ah, sorry I'll keep trying to get a repro then

**Comment by bradking** [comment]
Okay, thanks.  For reference, the corresponding discussion in CMake's issue is [here](https://gitlab.kitware.com/cmake/cmake/-/issues/22922#note_1086132).

**Comment by tylerjereddy** [comment]
Ok, this was way off target, we don't need directories to reproduce at all, I'll open another PR with some notes on the reproducer, it is quite sensitive to syntax details in the Fortran source.

**Pull request closed without merging**